test/support/warning_filter.rb

```diff
@@ -0,0 +1,53 @@
+require 'pathname'
+require 'set'
+
+module RegexpParserTest
+  # Stream filter for warnings
+  class WarningFilter
```

Comment:
@backus how would you feel about `WarningDetector`, `WarningCollector`, or something similar? Personally I don't like naming things `Observer`s that aren't an instance of the `Observer` pattern.